projects/angular-auth-oidc-client/src/lib/auto-login/auto-login-all-routes.guard.ts

```diff
@@ -17,39 +17,68 @@ export class AutoLoginAllRoutesGuard  {
     private readonly router: Router
   ) {}
 
-  canLoad(): Observable<boolean | UrlTree> {
-    return this.checkAuth(this.router.getCurrentNavigation()?.extractedUrl.toString().substring(1) ?? '');
+  canLoad(): Observable<boolean> {
+    const url = this.router.getCurrentNavigation()?.extractedUrl.toString().substring(1) ?? '';
+
+    return checkAuth(url, this.configurationService, this.checkAuthService, this.autoLoginService, this.loginService);
   }
 
   canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean | UrlTree> {
-    return this.checkAuth(state.url);
+    return checkAuth(state.url, this.configurationService, this.checkAuthService, this.autoLoginService, this.loginService);
   }
 
   canActivateChild(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean | UrlTree> {
-    return this.checkAuth(state.url);
+    return checkAuth(state.url, this.configurationService, this.checkAuthService, this.autoLoginService, this.loginService);
   }
+}
 
-  private checkAuth(url: string): Observable<boolean> {
-    return this.configurationService.getOpenIDConfiguration().pipe(
-      switchMap((config) => {
-        const allConfigs = this.configurationService.getAllConfigurations();
-
-        return this.checkAuthService.checkAuth(config, allConfigs).pipe(
-          take(1),
-          map(({ isAuthenticated }) => {
-            if (isAuthenticated) {
-              this.autoLoginService.checkSavedRedirectRouteAndNavigate(config);
-            }
-
-            if (!isAuthenticated) {
-              this.autoLoginService.saveRedirectRoute(config, url);
-              this.loginService.login(config);
-            }
-
-            return isAuthenticated;
-          })
-        );
-      })
-    );
-  }
+export function autoLoginAllRoutesGuard(): Observable<boolean> {
+  const configurationService = inject(ConfigurationService);
+  const checkAuthService = inject(CheckAuthService);
+  const loginService = inject(LoginService);
+  const autoLoginService = inject(AutoLoginService);
+  const router = inject(Router);
+
+  const url = router.getCurrentNavigation()?.extractedUrl.toString().substring(1) ?? '';
+
+  return checkAuth(url, configurationService, checkAuthService, autoLoginService, loginService);
+}
+
+// export function autoLoginAllRoutesGuard(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
```

Comment:
Originally I wanted to create a `ActivateGuard` and a `MatchGuard`, but I don't think this is necessary and it also caused some problems to test this guard (because it requires the `route` and `state` arguments).
This applies to both guards.

Because I spent some time on this I left this here in comments to discuss it before removing it or taking a closer look at the differences/tests.

Reply:
That sounds good. Thanks!